eng/common/scripts/SemVer.ps1

```diff
@@ -31,10 +32,7 @@ class AzureEngSemanticVersion {
     }
     
     AzureEngSemanticVersion([string] $versionString){
-        # Regex inspired but simplifie from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
-        $SEMVER_REGEX = "^(?<major>0|[1-9]\d*)\.(?<minor>0|[1-9]\d*)\.(?<patch>0|[1-9]\d*)(?:-?(?<prelabel>[a-zA-Z-]*)(?:\.?(?<prenumber>0|[1-9]\d*)))?$"
-        
-        if ($versionString -match $SEMVER_REGEX) {
+        if ($versionString -match "^$([AzureEngSemanticVersion]::SEMVER_REGEX)$") {
```

Comment:
Out of curiosity did you get a chance to validate this? I'm not sure if PS will like the unescaped `$` at the end.